File: demos/java/lambda-demo/src/main/java/com/mahendra/LambdaHandler.java
package com.mahendra;

import java.util.logging.Logger;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.amazonaws.services.lambda.runtime.events.S3Event;
import com.amazonaws.services.lambda.runtime.events.models.s3.S3EventNotification.S3ObjectEntity;

/**
 * This is a simple AWS Lambda function handler that processes input and returns a greeting message.
 * It implements the RequestHandler interface to handle incoming requests.
 */
public class LambdaHandler implements RequestHandler<S3Event, String> {

	private static final Logger log = Logger.getLogger(LambdaHandler.class.getName());
	@Override
	public String handleRequest(S3Event input, Context context) {
		
		S3ObjectEntity obj = input.getRecords().get(0).getS3().getObject();
		log.info(() -> "Starting Lambda function ....");
		log.info(() -> "Processing object: " + obj.getKey());
		log.info(() -> "Object size: " + obj.getSizeAsLong());
		

		log.info(() -> "End of function ....");
		return "Hello from AWS Lambda! Input: " + input;
	}
	
}

File: demos/python/lambda-demo/lambda_handler.py
import logging

def lambda_handler(event, context):
    logger = logging.getLogger()
    logger.setLevel(logging.INFO)

    # Assuming the event is an S3Event from AWS Lambda
    record = event['Records'][0]
    s3_object = record['s3']['object']
    key = s3_object['key']
    size = s3_object.get('size', 0)

    logger.info("Starting Lambda function ....")
    logger.info(f"Processing object: {key}")
    logger.info(f"Object size: {size}")
    logger.info("End of function ....")

    return f"Hello from AWS Lambda! Input: {event}"
